tests/api/test_token_usage_routes.py

```
@@ -0,0 +1,136 @@
+
+def _message(category: MessageCategory, tokens: int) -> Message:
+    return Message(role="user", content="x", category=category, tokens=tokens)
+
+
+def _session(session_id: str, tokens: int) -> Session:
+    session = Session(id=session_id)
+    session.messages.append(_message(MessageCategory.DIALOG, tokens))
+    return session
+
+
+def _core(sessions: list[Session]) -> Any:
```

Ruff (0.15.13)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `_core`

(ANN401)